### PR TITLE
chore!: migrate yaru.dart and yaru_icons.dart into yaru_widgets 

### PR DESCRIPTION
**Pull request by Feichtmeier**
⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ 

BREAKING CHANGE 

This migrates yaru.dart and yaru_icons.dart into this repository, but all commits are kept 🤞 

Bildschirmfoto 2024-02-22 um 17 31 19

Bildschirmfoto 2024-02-22 um 17 47 26

To merge this we need to check very precicly if everything is correctly merged

TODO

- [x] Check if the icons work correcty @Jupi007 (I moved your example to "YaruIcons" page in the example)
- [x] Check if the theme works correctly @Feichtmeier 
- [x] Double check if I didn't make any mistake @spydon and/or @d-loose 

Ref #835 

⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️

### CODE REVIEW

**Comment by Feichtmeier** [comment]
https://github.com/ubuntu/yaru_widgets.dart/assets/15329494/a84e100d-5a88-4e96-9317-402a97c94439

The goldens pass locally after clean/upgrade and regeneration
So they will prbly need the nightly CI to pass again here, so you can ignore those failed golden tests

File: pubspec.yaml
Comment (Jupi007) [inline pubspec.yaml:1]: You forgotten to include the font files:

```
flutter:
  fonts:
    - family: Ubuntu
      fonts:
        - asset: assets/fonts/Ubuntu-L.ttf
          weight: 300
        - asset: assets/fonts/Ubuntu-LI.ttf
          weight: 300
          style: italic
        - asset: assets/fonts/Ubuntu-R.ttf
          weight: 400
        - asset: assets/fonts/Ubuntu-RI.ttf
          weight: 400
          style: italic
        - asset: assets/fonts/Ubuntu-M.ttf
          weight: 500
        - asset: assets/fonts/Ubuntu-MI.ttf
          weight: 500
          style: italic
        - asset: assets/fonts/Ubuntu-B.ttf
          weight: 700
        - asset: assets/fonts/Ubuntu-BI.ttf
          weight: 700
          style: italic
`

**Review by Jupi007 — CHANGES_REQUESTED** [review]
I think we can improve and simplify some files placements: 
- `colors.dart` could be moved to the `foundation/` folder ;
- `icons_foundation/` could be merged with `foundation` ;
- `text/font/` should be moved to `assets/`, and `text/text_theme.dart` to `themes/` ;
- `icons_widgets/` to `widgets` ;
- and I'm not sure, but `theme_widgets/inherited_theme.dart` can maybe be moved to `widgets/` (or `themes/`) ?

Except that, this looks great. I didn't noticed any problem with icons :D

File: pubspec.yaml
Comment (Feichtmeier) [reply pubspec.yaml:1]: Oh god 🙇 thanks

**Comment by Feichtmeier** [comment]
> I think we can improve and simplify some files placements:
> 
>     * `colors.dart` could be moved to the `foundation/` folder ;
> 
>     * `icons_foundation/` could be merged with `foundation` ;
> 
>     * `text/font/` should be moved to `assets/`, and `text/text_theme.dart` to `themes/` ;
> 
>     * `icons_widgets/` to `widgets` ;
> 
>     * and I'm not sure, but `theme_widgets/inherited_theme.dart` can maybe be moved to `widgets/` (or `themes/`) ?
> 
> 
> Except that, this looks great. I didn't noticed any problem with icons :D

I kept the sub structure for this PR to make a first atempt where everything works. Maybe I could do this in a second PR? then I change from Fix to Ref in the PR desc
And thank you very much for catching the font!

**Comment by Jupi007** [comment]
> Maybe I could do this in a second PR?

Ofc, sounds good :+1:

**Review by Jupi007 — APPROVED** [review]
LGTM :+1:

File: README.md [hunk 1]
```diff
@@ -1,32 +1,60 @@
...
+# The official Flutter Yaru Theme and Widgets Suite
 
-[![Pub Package](https://img.shields.io/pub/v/yaru_widgets.svg)](https://pub.dev/packages/yaru_widgets)
+[![Pub Package](https://img.shields.io/pub/v/yaru_widgets.svg)](https://pub.dev/packages/yaru)
+
+
+This repository and package contains:
+
+- flutter widgets useful for building desktop and web applications, following but also expanding the yaru theme for the gnome desktop in Ubuntu 22.04+.
+- a theme for widgets from material.dart and from this repository here
```
Comment (spydon) [inline README.md]: This sentence reads a bit strange, maybe just remove "here"?

**Review by spydon — APPROVED** [review]
Lgtm!

**Pull request merged**